File: ShadowFrameworkJS/src/shadow/pipeline/expression/SFFunctionOperator.js

function SFFunctionOperator(func){
	this.list=new Array();
	this.setElement(func);
	this.setType(SFParameteri_GLOBAL_FLOAT);
	this.operatorSymbol="func";
}


inherit(SFFunctionOperator,SFExpressionOperator);



SFFunctionOperator.prototype["evaluateType"]=function(){
			this.updateSubExpressions();
			if(this.getElement()==="sample")
				this.setType(SFParameteri_GLOBAL_FLOAT4);
			else if(this.getElement()==="clamp")
				this.setType(this.list[0].getType());
			else if(this.getElement()==="normalize")
				this.setType(this.list[0].getType());
			else if(this.getElement()==="cross")
				this.setType(this.list[0].getType());
			else
				this.setType(SFParameteri_GLOBAL_FLOAT);
		};

	
SFFunctionOperator.prototype["cloneOperator"]=function(){
		return new SFFunctionOperator(getElement());
		};

SFFunctionOperator.prototype["evaluate"]=function(values){
			throw  ("Yet not implemented");
		};

File: ShadowFrameworkLite20/src/shadow/pipeline/expression/SFFunctionOperator.java
package shadow.pipeline.expression;

import shadow.math.SFValuenf;
import shadow.pipeline.parameters.SFParameteri;
import shadow.system.SFException;

public class SFFunctionOperator extends SFExpressionOperator{

	
	public SFFunctionOperator(String function) {
		super(function, SIZE_ALL);
	}

	@Override
	public void evaluateType()  throws SFExpressionException{
		updateSubExpressions();
		if(getElement().equalsIgnoreCase("sample"))
			setType(SFParameteri.GLOBAL_FLOAT4);
		else if(getElement().equalsIgnoreCase("clamp"))
			this.setType(list.get(0).getType());
		else if(getElement().equalsIgnoreCase("normalize"))
			this.setType(list.get(0).getType());
		else if(getElement().equalsIgnoreCase("cross"))
			this.setType(list.get(0).getType());
		else
			this.setType(SFParameteri.GLOBAL_FLOAT);
	}
	
	protected SFExpressionOperator cloneOperator() {
		return new SFFunctionOperator(getElement());
		
	}
	
//	@Override
//	public SFValuenf evaluate(SFValuesMap values) {
//		SFValuenf value1=getExpressionElement(0).evaluate(values);
//		SFValuenf value2=getExpressionElement(1).evaluate(values);
//		return new SFValue1f(value1.dot(value2));
//	}
	
	@Override
	public SFValuenf evaluate(SFExpressionValuesList values) {
		throw new SFException("Yet not implemented");
	}
}
